azure-pipelines.yml

```diff
@@ -20,10 +20,15 @@ steps:
     #
     # Azure
     #
-    Write-Host "`n`n$line`nAzure`n$line`n"
-    Invoke-RestMethod -Headers @{'Metadata'='true'} -URI http://169.254.169.254/metadata/instance?api-version=2017-12-01 -Method Get |
-      ConvertTo-Json -Depth 10 |
-      Write-Host
+    $assetTag = Get-WmiObject -class Win32_SystemEnclosure -namespace root\CIMV2 |
+      Select -ExpandProperty SMBIOSAssetTag
+    if ($assetTag -eq '7783-7084-3265-9085-8269-3286-77')
```

Comment:
This is a magic BIOS value that indicates Azure... go figure!

Reply:
Dear god.

Why not look for the existence of that `TM_BUILD` environment variable like we did in the `build.ps1`?

Reply:
Yeah, can we not use this same pattern: https://github.com/puppetlabs/pe-console-services/blob/2019.1.x/docker/ci/build.ps1#L11-L14

Reply:
Because that won't work. `TF_BUILD` is always set when you're running the VSTS agent for a build - it doesn't matter where the VM lives.

We need to know whether we're actually on an Azure VM - and this is the only documented way of verifying that.